lnrpc/rpc.proto

```diff
@@ -1006,6 +1006,19 @@ message HTLC {
     int64 amount = 2;
     bytes hash_lock = 3;
     uint32 expiration_height = 4;
+
+    // Index identifying the htlc on the channel.
+    uint64 htlc_index = 5;
+
+    // If this HTLC is involved in a forwarding operation, this field indicates
+    // the linked channel. For an outgoing htlc, it is the incoming channel. For
+    // an incoming htlc, it is the outgoing channel. When the htlc originates
+    // from this node or this node is the final destination, the linked channel
+    // will be zero.
+    uint64 linked_channel = 6;
```

Comment:
Not sure about the naming on this, what about `forwarding_channel` + `forwarding_htlc_index`? I think that makes sense in the context of an incoming or outgoing htlc, and for our own payments/receives. The word "linked" doesn't have much meaning for an external user (I'd think).

Reply:
I was struggling with the name. Will rename to `forwarding_*`